compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -1469,6 +1486,13 @@ fn generic_simd_intrinsic<'ll, 'tcx>(
         }
         return Ok(bx.extract_element(args[0].immediate(), bx.const_i32(idx as i32)));
     }
+    if name == sym::simd_extract_dyn {
+        require!(
+            ret_ty == in_elem,
+            InvalidMonomorphization::ReturnType { span, name, in_elem, in_ty, ret_ty }
+        );
+        return Ok(bx.extract_element(args[0].immediate(), args[1].immediate()));
```

Comment:
So, eh, this generates the exact same LLVM IR as `simd_extract`, and in Miri it's also the same. Why does this have to be a separate intrinsic? If we just lift the restriction for the argument to be a constant, shouldn't that work?

Reply:
We would like to have separate const and dynamic intrinsics here to make it easier to guarantee certain codegen, especially for backends which will have to handle them differently.

Reply:
I feel like that's something cranelift should handle internally rather than leaking that to the rest of the compiler & language.

Obviously we should have codegen tests to ensure that when the idx is `const { ... }`, it generates the code we want it to generate, but we need those tests anyway so I don't see how that's an argument for the extra complexity of having two intrinsics and having some super-special does-not-exist-anywhere-else-in-the-typesystem ad-hoc hacks to ensure the argument is indeed a constant. The more special cases like that we can remove the better. :)  (We have a type system way of expressing "must be const": use a const generic parameter.)

Reply:
I mean, I'd be happy with just having `simd_insert` and `simd_extract` be const generic functions which call to their `_dyn` variant by their default implementation, and then the intrinsicness of it is only for overriding purposes.